### PR TITLE
Feature: Sockets Implementation for Notifications

### PR DESCRIPTION
**Pull request by jerixmx**
### What this PR does (required):
- Updates notifications (both dropdown and Notifications page) when user submits a design.

### Screenshots / Videos (required):
https://user-images.githubusercontent.com/78225194/127560785-6e9ba19a-4f0c-465d-9d0c-bebd470da837.mp4

### Any information needed to test this feature (required):
- Check whether notifications are available upon logging in.
- Upload/submit an image then access the Notifications dropdown and Notifications page. These should be updated without having to refresh the page.

### Any issues with the current functionality (optional):
- Triggering notification creation for new messages is not yet implemented while Messaging is not yet available.
- May have to identify notification receiver when emitting "notification created" from the server.

### CODE REVIEW

File: server/controllers/notification.js [hunk 1]
```diff
@@ -64,3 +64,37 @@ exports.getNotifications = asyncHandler(async (req, res, next) => {
...
+      }
+
+      const notification = await Notification.create(params);
+
+      if (!notification) {
+        throw new Error('Invalid notification data');
+      }
+    }
+
+    socket.emit('notification created');
```
Comment (rajivtitus) [inline server/controllers/notification.js]: You need to send the notification through an individual socket id :)


Cheatsheet: https://socket.io/docs/v3/emit-cheatsheet/index.html
Tutorial: https://www.youtube.com/watch?v=HggSXt1Hzfk&t=5312s (Watch the part where sockets are being implemented but this is for messaging)

Comment (jerixmx) [reply server/controllers/notification.js]: Thanks for the references!

Comment (eric-silva-61) [reply server/controllers/notification.js]: I know this is super late, but if you look at my PR I have the socketId collection available and maintained in app.js
69-74 is receiving an incoming message.  It takes the email (send up as receiver) and uses the helper getUser to get an object with the email and socketId.  
```socket.on('SEND_MESSAGE', ({ receiver, conversationId, message }) => {
    const receiverId = getUser(receiver);
    if (receiverId) {
      io.to(receiverId.socketId).emit('GET_MESSAGE', { conversationId, message });
    }
  });```

Comment (jerixmx) [reply server/controllers/notification.js]: Thanks, this helps a lot! I tried to implement something similar just now. We can check later if that works alright.

**Review by rajivtitus — APPROVED** [review]

**Review by rajivtitus — COMMENTED** [review]
LGTM!

**Pull request merged**